book/src/app.md

```diff
@@ -1,54 +1,33 @@
 # Application
 
-`App` define the type of note (e.g. XAN, ETH, BTC). Each application is identified by an address `appAddress` (the same way as user address identifies a user) and has its own VP `AppVP`.
+### Application logic
+Application VP is the defining structure of an application that contains the application logic. 
+Every time the application state is about to change, the application VP is called to validate the state transition.
+If the application VP doesn't consider a state transition valid, the state will not be changed.
 
-### Application VP
-Each application has its own [validity predicate](./validity-predicates.md) `AppVP` that defines the conditions on which the application can be sent/received/etc (e.g. whitelist VP that only allows using the application a specified set of users). As with other VPs, `AppVP` checks that input and output notes of the tx satisfy certain constraints.
-It is required that the `AppVP` of the applications involved in a tx evaluated to `true`.
+An application VP might require validity of other VPs, enforcing the VP hierarchy with the application VP being the main VP checked.
 
-In Taiga, VPs are shielded, so instead of showing that `AppVP` evaluates to `true` publicly, a ZK proof is created. To make sure that `AppVP`  evaluates to `true`, an observer can verify the proof (using the verifier key):
+#### Application address
+Each application has an address that identifies the application. The application address is derived from its application VP (or VPs, if there is more than one).
 
-```verify(AppVP_proof, app_VK) = True```
+### Application state
+As Taiga works in the UTXO model, application states are stored in notes, and changing the application state is done by spending the old application state notes and creating new application state notes.
 
-### Application Address
-Each app is identified by an address that is derived from its verifier key `app_VK`:
-`appAddress = Com(app_VK)`
+The application a note belongs to is indicated by the note's type which is derived using the application address, binding the note with the application.
+One application can have notes of multiple types, all of which are derived using the application address and some additional information that helps to distinguish the application types.
```

Comment:
I'm not sure I understand this. Do you mean that a note can access the other 3 notes from which the transaction is executed?

Reply:
An application A can have notes of types AA, AB, AC, etc. All of them are binded to the application A, but they have distinct types and don't balance each other. They are independent as if they were notes of different applications, but they are actually binded to the same application. For example, intent notes are like that. The value base of intent notes are derived using userVPs, so they are different for each userVP, but they still belong to the same intent application.